### PR TITLE
Add the GET /api/topics endpoint

### PR DESCRIPTION
**Pull request by IanFindlay**
As the first endpoint, this PR also:
- Adds express and supertest to the package files
- Sets up the testing environment
- Initialises the app
- Creates data/index.js for environment-based selection of which data to use

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                  slug: expect.any(String),
+                  description: expect.any(String),
+                })
+              );
+            });
+          });
+      });
+    });
+  });
+});
```
Comment (Angaca) [inline __tests__/app.test.js:42]: Good testing

File: models/topics.models.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+const db = require("../db/connection");
+
+exports.selectTopics = () => {
+  return db.query(`SELECT * FROM topics`).then(({ rows: topics }) => topics);
```
Comment (Angaca) [inline models/topics.models.js]: Be careful, you are missing the ; after topics

File: controllers/errors.controllers.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+exports.handleInvalidEndpoint = (req, res) => {
+  res.status(404).send({ msg: "Path not found" });
+};
```
Comment (Angaca) [inline controllers/errors.controllers.js:3]: Great that you separated errors in a single file

**Review by Angaca — APPROVED** [review]
Good to go!

File: models/topics.models.js (hunk 2 above)
Comment (IanFindlay) [reply models/topics.models.js]: Cheers for the review, will fix this then merge.

**Pull request merged**